src/transformers/models/informer/modeling_informer.py

```diff
@@ -831,11 +837,23 @@ def forward(
                 f" {attn_weights.size()}"
             )
 
-        # Original impl don't apply attention_mask to the encoder, only for the decoder
-        # For the decoder, it creates a casual mask sliced with M_top (ProbMask)
-        if self.is_decoder:
-            prob_mask = _prepare_decoder_prob_attention_mask(L=L_Q, M_top=M_top, scores=attn_weights)
-            attn_weights.masked_fill_(prob_mask, -np.inf)
+        if attention_mask is not None:
```

Comment:
By doing so, aren't we apply the prob mask to the encoder? because the original impl don't apply `attention_mask` to the encoder, only for the decoder

Reply:
Talked in Slack